dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1823,7 +1823,11 @@ private boolean resolveHang(long tasksCompleted, boolean queueEmpty, int poolSiz
                         // happens for a given hang, but otherwise just bail
                         if (hangMaxThreadsMessageEmitted == false && hangIntervalCounter > 0) {
                             if (tc.isWarningEnabled()) {
-                                Tr.warning(tc, "unbreakableExecutorHang", poolSizeWhenHangDetected, poolSize);
+                                if (poolSizeWhenHangDetected != poolSize) {
+                                    Tr.warning(tc, "unbreakableExecutorHang", poolSizeWhenHangDetected, poolSize);
+                                } else {
+                                    Tr.warning(tc, "unbreakableExecutorHang2", poolSize);
```

Comment:
```suggestion
                                    Tr.warning(tc, "CWWKE1207.all.threads.hung", poolSize);
```

Reply:
I made the suggested change, with slightly different message key wording.